src/Renderer/Swapchain/Swapchain.h

```diff
@@ -4,58 +4,212 @@
+         */
         u32 GetImageCount() { return imageCount; }
+
+        /**
+         * @brief The the raw Vulkan swapchain struct. 
+         * 
+         * @return VkSwapchainKHR 
+         */
         VkSwapchainKHR GetSwapChain() { return swapChain; }
 
+        /**
+         * @brief Get the swapchain image width.
+         * 
+         * @return u32 representing the width of the image. 
+         */
         u32 GetWidth() { return swapChainExtent.width; }
+
+        /**
+         * @brief Get the swapchain image height.
+         * 
+         * @return u32 representing the image height.
+         */
         u32 GetHeight() { return swapChainExtent.height; }
 
+        /**
+         * @brief Get the current render pass object. 
+         * 
+         * @return VkRenderPass - the raw vulkan render pass object. 
+         */
         VkRenderPass GetRenderPass() { return renderPass; }
 
+        /**
+         * @brief Loads in the next image to be written to in the Swapchain. 
+         * 
+         * @param imageIndex - the index of the next image. 
+         * @return VkResult - the result of acquiring the next image. 
+         */
         VkResult AcquireNextImage(u32* imageIndex);
+
+        /**
+         * @brief Submits a command buffer for drawing.
+         * 
+         * @param buffers - an array of command buffers. 
+         * @param imageIndex - the index of the image being drawn. 
+         * @return VkResult - the result of submitting the buffer. 
+         */
         VkResult SubmitCommandBuffers(const VkCommandBuffer* buffers, u32* imageIndex);
 
         private:
         
+        /**
+         * @brief Create a Vulkan swapchain object. 
+         */
         void CreateSwapChain();
+
+        /**
+         * @brief Create image views for our images. An image view represents the subresources
+         * that images must have to be valid. An image view must be created for each image being drawn.
+         */
         void CreateImageViews();
+
+        /**
+         * @brief Creates a render pass. In Vulkan, render passes are objects that represent 
+         * the state that rendering resources must be in to render to an image. All image rendering
+         * must occur within a render pass, while compute operations do not. 
+         */
         void CreateRenderPass();
+
+        /**
+         * @brief Create a Depth Resources object. Depth resources represent how object depth is calculated.
+         * These resources are written to their own resource views. 
+         */
         void CreateDepthResources();
+
+        /**
+         * @brief Creates the swapchain's frame buffers. A frame buffer represents all the data needed to be submitted 
+         * to the GPU in a frame. 
+         */
         void CreateFrameBuffers();
+
+        /**
+         * @brief Inialises all thread synchronisation objects. Vulkan is multi-threaded and submits images
```

Comment:
```suggestion
         * @brief Initialises all thread synchronisation objects. Vulkan is multi-threaded and submits images
```

Reply:
I think you have spellcheck turned off in your IDE

Reply:
yeah you're right. I removed typos because it was giving me issues with variable names.